v2/event/event_unmarshal.go

```diff
@@ -177,7 +174,10 @@ func ReadJson(out *Event, reader io.Reader) error {
 
 		// If no specversion, enqueue unconditionally
 		if !checkFlag(state, specVersionV03Flag|specVersionV1Flag) {
-			tokenQueue.push(key, iterator.ReadAny())
+			if tokenQueue == nil {
+				tokenQueue = make([]entry, 0, preallocQueueSize)
+			}
+			tokenQueue = append(tokenQueue, entry{key: key, value: iterator.ReadAny()})
```

Comment:
I have to say that I'm very surprised! Using `append` is faster than assigning to preallocated memory with indexes?!

Reply:
It's not. It should be equally quick. The win here is the saved allocations.